src/lib/utilities/get-event-categorization.test.ts

```diff
@@ -0,0 +1,320 @@
+      eventTypeInCategory('workflow')({
+        eventType: 'WorkflowExecutionStarted',
+      }),
+    ).toBe(true);
+  });
+
+  it('should categorize a "WorkflowExecutionTerminated" as a "workflow"', () => {
+    expect(
+      eventTypeInCategory('workflow')({
+        eventType: 'WorkflowExecutionTerminated',
+      }),
+    ).toBe(true);
+  });
+
+  it('should categorize a "WorkflowExecutionTimedOut" as a "workflow"', () => {
+    expect(
+      eventTypeInCategory('workflow')({
+        eventType: 'WorkflowExecutionTimedOut',
+      }),
+    ).toBe(true);
+  });
+
+  it('should categorize a "WorkflowTaskCompleted" as a "workflow"', () => {
+    expect(
+      eventTypeInCategory('workflow')({ eventType: 'WorkflowTaskCompleted' }),
+    ).toBe(true);
+  });
+
+  it('should categorize a "WorkflowTaskFailed" as a "workflow"', () => {
+    expect(
+      eventTypeInCategory('workflow')({ eventType: 'WorkflowTaskFailed' }),
+    ).toBe(true);
+  });
+
+  it('should categorize a "WorkflowTaskScheduled" as a "workflow"', () => {
+    expect(
+      eventTypeInCategory('workflow')({ eventType: 'WorkflowTaskScheduled' }),
+    ).toBe(true);
+  });
+
+  it('should categorize a "WorkflowTaskStarted" as a "workflow"', () => {
+    expect(
+      eventTypeInCategory('workflow')({ eventType: 'WorkflowTaskStarted' }),
+    ).toBe(true);
+  });
+
+  it('should categorize a "WorkflowTaskTimedOut" as a "workflow"', () => {
+    expect(
+      eventTypeInCategory('workflow')({ eventType: 'WorkflowTaskTimedOut' }),
+    ).toBe(true);
+  });
+
+  it('should categorize a "ExternalWorkflowExecutionCancelRequested" as a "workflow"', () => {
+    expect(
+      eventTypeInCategory('workflow')({
+        eventType: 'ExternalWorkflowExecutionCancelRequested',
+      }),
+    ).toBe(true);
+  });
+
+  it('should categorize a "ExternalWorkflowExecutionSignaled" as a "workflow"', () => {
+    expect(
+      eventTypeInCategory('workflow')({
+        eventType: 'ExternalWorkflowExecutionSignaled',
+      }),
+    ).toBe(true);
+  });
+
+  it('should categorize a "RequestCancelExternalWorkflowExecutionFailed" as a "workflow"', () => {
+    expect(
+      eventTypeInCategory('workflow')({
+        eventType: 'RequestCancelExternalWorkflowExecutionFailed',
+      }),
+    ).toBe(true);
+  });
+
+  it('should categorize a "RequestCancelExternalWorkflowExecutionInitiated" as a "workflow"', () => {
+    expect(
+      eventTypeInCategory('workflow')({
+        eventType: 'RequestCancelExternalWorkflowExecutionInitiated',
+      }),
+    ).toBe(true);
+  });
+
+  it('should categorize a "MarkerRecorded" as a "command"', () => {
+    expect(
+      eventTypeInCategory('command')({ eventType: 'MarkerRecorded' }),
+    ).toBe(true);
+  });
+
+  it('should categorize a "UpsertWorkflowSearchAttributes" as a "command"', () => {
+    expect(
+      eventTypeInCategory('command')({
+        eventType: 'UpsertWorkflowSearchAttributes',
+      }),
+    ).toBe(true);
+  });
+});
```

Comment:
All of these are true it would probably be good to test the opposite of these functions

```
    expect(
      eventTypeInCategory('workflow')({
        eventType: 'UpsertWorkflowSearchAttributes',
      }),
    ).toBe(false);
    ```